aptos-move/framework/aptos-framework/sources/configs/stake.move

```diff
@@ -1062,8 +1077,8 @@ module aptos_framework::stake {
         // Here we do multiplication before division to minimize rounding errors.
         let base_rewards =
             voting_power * validator_set_config.rewards_rate / validator_set_config.rewards_rate_denominator;
-        let rewards_amount = base_rewards * num_successful_votes / num_blocks;
-        if (rewards_amount > 0) {
+        if (base_rewards > 0 && num_successful_proposals > 0) {
```

Comment:
Can you add an extra unit test for when num_successful = 0 but total_proposals > 0. This wasn't possible before as the number of blocks can never be 0.

Reply:
num_successful = 0 is the only way to get 0 rewards, and the updated version of previous test does check that.

I've did that by having new_epoch not add a new successful block, but make sure that each validator has at least one proposal. If they have a failed proposal - it leaves it as is. If it has neither successful nor failed proposal, it makes it have one successful.
So the test case where we set one failed proposal for the validator2, has num_successful = 0 and total_proposals=1